packages/asset-swapper/src/utils/market_operation_utils/comparison_price.ts

```diff
@@ -0,0 +1,79 @@
+import { Web3Wrapper } from '@0x/dev-utils';
+import { BigNumber, logUtils } from '@0x/utils';
+import * as _ from 'lodash';
+
+import { MarketOperation } from '../../types';
+
+import { COMPARISON_PRICE_DECIMALS } from './constants';
+import { ComparisonPrice, ERC20BridgeSource, FeeEstimate, FeeSchedule, MarketSideLiquidity } from './types';
+
+/**
+ * Takes in an optimizer response and returns a price for RFQT MMs to beat
+ * returns the price of the taker asset in terms of the maker asset
+ * So the RFQT MM should aim for a higher price
+ * @param adjustedRate the adjusted rate (accounting for fees) from the optimizer, maker/taker
+ * @param amount the amount specified by the client
+ * @param marketSideLiquidity the results from querying liquidity sources
+ * @param feeSchedule the fee schedule passed to the Optimizer
+ * @return ComparisonPrice object with the prices for RFQ MMs to beat
+ */
+export function getComparisonPrices(
+    adjustedRate: BigNumber,
+    amount: BigNumber,
+    marketSideLiquidity: MarketSideLiquidity,
+    feeSchedule: FeeSchedule,
+): ComparisonPrice {
+    let wholeOrder: BigNumber | undefined;
+    let feeInEth: BigNumber | number;
+
+    // HACK: get the fee penalty of a single 0x native order
+    // The FeeSchedule function takes in a `FillData` object and returns a fee estimate in ETH
+    // We don't have fill data here, we just want the cost of a single native order, so we pass in undefined
+    // This works because the feeSchedule returns a constant for Native orders, this will need
+    // to be tweaked if the feeSchedule for native orders uses the fillData passed in
+    // 2 potential issues: there is no native fee schedule or the fee schedule depends on fill data
+    if (feeSchedule[ERC20BridgeSource.Native] === undefined) {
+        logUtils.warn('ComparisonPrice function did not find native order fee schedule');
+
+        return { wholeOrder };
+    } else {
+        try {
+            feeInEth = new BigNumber((feeSchedule[ERC20BridgeSource.Native] as FeeEstimate)(undefined));
```

Comment:
When would this throw an error exactly? is there any way that we can avoid the try/catch?

Reply:
It would throw if the feeSchedule you specify for native orders depends on the content of the orders themselves. See protocol/packages/asset-swapper/src/utils/market_operation_utils/constants.ts. The default native order fee is just a constant, but it's possible to use the fill data in a function.